examples/full_feature/Application.cpp

```diff
@@ -0,0 +1,305 @@
+    }
+};
+
+/**
+ * This is interrupt logic used to stop the application from running.
+ */
+std::function<void(int)> sigintCall;
+
+void sigintResponse(int signal)
+{
+    if (sigintCall != nullptr)
+        sigintCall(signal);
+}
+
+int main(int /* argc */, char** /* argv */)
+{
+    /**
+     * Setting up the default console logger.
+     * For more info, increase the LogLevel.
+     * Logging to file could also be added here, by adding the type `Logger::Type::FILE` and passing a file path as
+     * third argument.
+     */
+    wolkabout::Logger::init(wolkabout::LogLevel::TRACE, wolkabout::Logger::Type::CONSOLE);
+
+    /**
+     * Now we can create the device using the user provided device credentials.
+     * We will also create the data object, in which we store the state of all values of the device.
+     * And the persistence object, that will be storing the values in cold-storage, that would be available to us at any
+     * moment. There are only two values that the platform could send us - Switch ("SW"), or Heartbeat ("HB"), so only
+     * those two will be stored by the persistence.
+     */
+    auto device = wolkabout::Device{DEVICE_KEY, DEVICE_PASSWORD, wolkabout::OutboundDataMode::PUSH};
+    auto deviceInfo = DeviceData{0, false, std::chrono::seconds(60)};
+    auto deviceInfoHandler = std::make_shared<DeviceDataChangeHandler>(deviceInfo);
+
+    /**
+     * Now we can start creating the Wolk instance that is right for us.
+     * We will also create some in memory persistence so messages can get buffered if the platform connection gets
+     * interrupted.
+     */
+    auto inMemoryPersistence = std::make_shared<wolkabout::InMemoryPersistence>();
+    auto wolk = wolkabout::WolkBuilder(device)
+                  .host(PLATFORM_HOST)
+                  .ca_cert_path(CA_CERT_PATH)
+                  .feedUpdateHandler(deviceInfoHandler)
+                  .withPersistence(inMemoryPersistence)
+                  .withFileTransfer(FILE_MANAGEMENT_LOCATION)
+                  // Uncomment for FileURLDownload
+                  //                  .withFileURLDownload(FILE_MANAGEMENT_LOCATION, nullptr, true)
+                  // Uncomment for a FileListener
+                  //                  .withFileListener(std::make_shared<ExampleFileListener>())
+                  .withFirmwareUpdate(std::make_shared<ExampleFirmwareInstaller>(FILE_MANAGEMENT_LOCATION))
+                  // Uncomment for example ParameterListener
+                  //                  .withFirmwareUpdate(std::make_shared<ExampleFirmwareParameterListener>())
+                  .build();
+
+    /**
+     * Now we can start the running logic of the connector. We will connect to the platform, and start running a loop
+     * that will update the temperature feed. This loop will also be affected by the heartbeat feed.
+     */
+    wolk->connect();
+    bool running = true;
+    sigintCall = [&](int signal) {
```

Comment:
You cannot run arbitrary code in signal handler https://en.cppreference.com/w/cpp/utility/program/signal
Just set the flag to exit the loop and do the disconnect after loop